internal/bminventory/inventory.go

```diff
@@ -4429,6 +4429,10 @@ func (b *bareMetalInventory) uploadLogs(ctx context.Context, params installer.Up
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 	if params.LogsType == string(models.LogsTypeController) {
+		firstClusterLogCollectionEvent := false
+		if swag.IsZero(currentCluster.ControllerLogsCollectedAt) {
```

Comment:
What do you think of moving this logic right before the check in line 4446? I was also wondering that if you're only going to use this variable once, why not just use the comparison in the conditional directly in line 4446, rather than declare a variable.

Reply:
If I do that, the call to `b.clusterApi.SetUploadControllerLogsAt` will place a timestamp in `ControllerLogsCollectedAt` and I'll have no way to know this was done for the first time.

Reply:
My proposal was to do this:
```go
    if params.LogsType == string(models.LogsTypeController) {
                err = b.clusterApi.SetUploadControllerLogsAt(ctx, currentCluster, b.db)
		if err != nil {
			log.WithError(err).Errorf("Failed update cluster %s controller_logs_collected_at flag", params.ClusterID)
			return common.NewApiError(http.StatusInternalServerError, err)
		}
		err = b.clusterApi.UpdateLogsProgress(ctx, currentCluster, string(models.LogsStateCollecting))
		if err != nil {
			log.WithError(err).Errorf("Failed update cluster %s log progress %s", params.ClusterID, string(models.LogsStateCollecting))
			return common.NewApiError(http.StatusInternalServerError, err)
		}
		if swag.IsZero(currentCluster.ControllerLogsCollectedAt) {
		   b.eventsHandler.AddEvent(ctx, params.ClusterID, nil, models.EventSeverityInfo,
				fmt.Sprintf("Uploaded logs for cluster %s",
					params.ClusterID.String()), time.Now())
		}
    }		

```

Does it make sense?
Also, I'm not seeing how the definition of `firstclusterLogCollectionEvent` is impacting the call to `b.clusterApi.SetUploadConctrollerLogsAt`. What am I missing?

Reply:
@jordigilh thank you for taking the time for this and the feedback.
I'll try to better elaborate: 

The idea behind this PR is to issue a single Event for we collect controller logs.
We need to determine that starting from the second logs upload onward, the backend should not publish an event.

The way to do this is to inspect the `ControllerLogsCollectedAt` timestamp, which will return True for `swag.IsZero` if it was never set (and it does not in its initial state). Therefore, I placed that first if statement before the call to `b.clusterApi.SetUploadControllerLogsAt`, which will tell us just that.

The `b.clusterApi.SetUploadControllerLogsAt` function will alter that timestamp:

https://github.com/openshift/assisted-service/blob/0daa74d48c2fd1d94864c9594ca2aa8a93863016/internal/cluster/cluster.go#L392-L398

So with your proposal, I suspect if `swag.IsZero(currentCluster.ControllerLogsCollectedAt)`
will always return false, hence no event at all.

Would you please let me know if I missed something here?

Reply:
Makes sense. Thanks for the clarification! 🙂